andela/casablanca-rc
# #157032079 Add reviews and ratings

#### What does this PR do?
add reviews and ratings for shops and products
#### Description of Task to be completed?
* create a custom plugin for product reviews
* register the plugin
* create components for product reviews
* create meteor methods for product reviews
* create publications and subscriptions for product reviews
* create a custom plugin for shop reviews
* register the plugin
* create components for shop reviews
* create meteor methods for shop reviews
* create publications and subscriptions for shop reviews
* add the name of the custom plugin for shop-reviews to the defaultVisitorRoles array in `/server/core/core.js`
#### How should this be manually tested?
* clone this repo
* cd into the project folder
* run `meteor npm install` to install project dependecies
* install `ImageMagick`
* run `reaction` to start up the app
* click on `Sign In` at the top right corner and register
* click on `Guest` at the top right corner
* click on `Profile` and edit your details and save
* click on `Reaction` at the top left corner
* click on the product on the page
* click on the `Add Review` modal at the bottom of the page
* click on a star to rate and give a review
* click `Add Review` to save
* you should see a review under the modal
* click on the `Reaction` link under the product title
* click on the `Add Review` modal at the right side of the new page
* click on a star to rate and also give a review
* click `Add Review` to save
* you should see a review under the modal
#### Any background context you want to provide?
neither reviews or ratings are optional and both need to be provided
#### What are the relevant pivotal tracker stories?(if applicable)
#157032079
